packages/adblocker/src/resources.ts

```diff
@@ -38,112 +130,166 @@ export default class Resources {
+        names,
+        content,
+        executionWorld,
+        requiresTrust,
+        dependencies,
+      };
+      for (const name of names) {
+        scriptlets.set(name, scriptlet);
+      }
+    }
 
     return new Resources({
       checksum,
-      js,
       resources,
+      scriptlets,
+      js: scriptletsToJsMapping(scriptlets),
     });
   }
 
   public static parse(data: string, { checksum }: { checksum: string }): Resources {
-    const typeToResource: Map<string, Map<string, string>> = new Map();
-    const trimComments = (str: string) => str.replace(/^\s*#.*$/gm, '');
-    const chunks = data.split('\n\n');
-
-    for (const chunk of chunks) {
-      const resource = trimComments(chunk).trim();
-      if (resource.length !== 0) {
-        const firstNewLine = resource.indexOf('\n');
-        const split = resource.slice(0, firstNewLine).split(/\s+/);
-        const name = split[0];
-        const type = split[1];
-        const body = resource.slice(firstNewLine + 1);
-
-        if (name === undefined || type === undefined || body === undefined) {
-          continue;
-        }
-
-        let resources = typeToResource.get(type);
-        if (resources === undefined) {
-          resources = new Map();
-          typeToResource.set(type, resources);
-        }
-        resources.set(name, body);
+    const distribution: ResourcesDistribution = JSON.parse(data);
+    const resources: Map<string, RedirectResource> = new Map();
+
+    for (const redirect of distribution.redirects) {
+      for (const name of redirect.names) {
+        resources.set(name, redirect);
       }
     }
 
-    // The resource containing javascirpts to be injected
-    const js: Map<string, string> = typeToResource.get('application/javascript') || new Map();
-    for (const [key, value] of js.entries()) {
-      if (key.endsWith('.js')) {
-        js.set(key.slice(0, -3), value);
+    const scriptlets: Map<string, Scriptlet> = new Map();
+    for (const scriptlet of distribution.scriptlets) {
+      for (const name of scriptlet.names) {
+        scriptlets.set(name, scriptlet);
       }
     }
 
-    // Create a mapping from resource name to { contentType, data }
-    // used for request redirection.
-    const resourcesByName: Map<string, Resource> = new Map();
-    typeToResource.forEach((resources, contentType) => {
-      resources.forEach((resource: string, name: string) => {
-        resourcesByName.set(name, {
-          contentType,
-          body: resource,
-        });
-      });
-    });
-
     return new Resources({
       checksum,
-      js,
-      resources: resourcesByName,
+      scriptlets,
+      resources,
+      js: scriptletsToJsMapping(scriptlets),
     });
   }
 
   public readonly checksum: string;
   public readonly js: Map<string, string>;
-  public readonly resources: Map<string, Resource>;
+  public readonly scriptlets: Map<string, Scriptlet>;
+  public readonly resources: Map<string, RedirectResource>;
```

Comment:
having both `js` and `resources` will be confusing. Lets break backwards compatibility and make `js` to be private, maybe we can rename it to `#scripletCache`. 

Lets also introduce `getScriptlet` method, which will be similar to `getResource` and will become the primary way to access the assembled scriplets.

Reply:
Renamed to `scriptletCaches` but didn't drop or change `scriptlets` as we currently get scriptlet name in the filter level.